src/legacy/core_plugins/metrics/public/components/vis_editor_visualization.js

```diff
@@ -128,19 +140,42 @@ class VisEditorVisualization extends Component {
       this.setState((prevState) => {
         const newHeight = prevState.height + (keyCode === keyCodes.UP ? -15 : 15);
         return {
-          height: Math.max(MIN_CHART_HEIGHT, newHeight)
+          height: Math.max(MIN_CHART_HEIGHT, newHeight),
         };
       });
     }
   }
 
+  hasShowPanelIntervalValue() {
+    const type = get(this.props, 'model.type', '');
+
+    return [
+      'metric',
+      'top_n',
+      'gauge',
+      'markdown'
+    ].includes(type);
+  }
+
+  getFormattedPanelInterval() {
+    const interval = get(this.props, 'model.interval', 0);
```

Comment:
why `0` as default? It's better to keep the interval type consistent so we can easily convert it then to TS with no hassle on understanding why this is a `string|number` where we don't have any converter for numbers. If is just for the next `if` check please use `undefined`.